src/Api.php

```diff
@@ -206,33 +212,81 @@ public function delete($pattern, $callable = null)
     /**
      * Implement REST pattern matching.
      *
-     * @param string           $pattern
-     * @param \atk4\data\Model $model
+     * @param string                    $pattern
+     * @param \atk4\data\Model|callable $model
      *
      * @return mixed
      */
     public function rest($pattern, $model = null)
     {
+        // GET all records
         $this->get($pattern, function () use ($model) {
+            $args = func_get_args();
+
+            if (is_callable($model)) {
+                $model = call_user_func_array($model, $args);
+            }
+
             return $model;
         });
 
-        $this->get($pattern.'/:id', function ($id) use ($model) {
+        // GET :id - one record
+        $this->get($pattern.'/:id', function () use ($model) {
+            $args = func_get_args();
+            $id = array_pop($args); // pop last element of args array, it's :id
+
+            if (is_callable($model)) {
+                $model = call_user_func_array($model, $args);
+            }
+
             return $model->load($id)->get();
         });
 
-        $this->patch($pattern.'/:id', function ($id) use ($model) {
-            return $model->load($id)->set($this->requestData)->save()->get();
+        // PATCH :id - update one record (same as POST :id)
+        $this->patch($pattern.'/:id', function () use ($model) {
+            $args = func_get_args();
+            $id = array_pop($args); // pop last element of args array, it's :id
+
+            if (is_callable($model)) {
+                $model = call_user_func_array($model, $args);
+            }
+
+            return $model->load($id)->save($this->requestData)->get();
         });
-        $this->post($pattern.'/:id', function ($id) use ($model) {
-            return $model->load($id)->set($this->requestData)->save()->get();
+
+        // POST :id - update one record
+        $this->post($pattern.'/:id', function () use ($model) {
+            $args = func_get_args();
+            $id = array_pop($args); // pop last element of args array, it's :id
+
+            if (is_callable($model)) {
+                $model = call_user_func_array($model, $args);
+            }
+
+            return $model->load($id)->save($this->requestData)->get();
         });
-        $this->delete($pattern.'/:id', function ($id) use ($model) {
+
+        // DELETE :id - delete one record
+        $this->delete($pattern.'/:id', function () use ($model) {
+            $args = func_get_args();
+            $id = array_pop($args); // pop last element of args array, it's :id
```

Comment:
lets use $args['id'] , because order my be messed up.

Reply:
you'd have to amend the callback function format.

Reply:
order will not be messed up because this :id is last argument added to $vars array (in match() method). Name of arguments in pattern doesn't matter - only order of them matters.

Reply:
i still prefer if it's a named array, i think it's better long-term.